src/fdmdv2_main.cpp

```diff
@@ -1163,70 +1195,72 @@ void MainFrame::OnTimer(wxTimerEvent &evt)
-
-                wxString s;
-                if (ret && (checksum_tx == checksum_rx)) {
-                    m_callsign[strlen(m_callsign)-2] = 0;
-                    s.Printf("%s", m_callsign);
-                    m_txtCtrlCallSign->SetValue(s);
-
-#ifdef __UDP_EXPERIMENTAL__
-                    char s1[MAX_CALLSIGN];
-                    sprintf(s1,"rx_txtmsg %s", m_callsign);
-                    processTxtEvent(s1);
-
-                    m_checksumGood++;
-                    s.Printf("%d", m_checksumGood);
-                    m_txtChecksumGood->SetLabel(s);              
-#endif
-                }
-                else {
-#ifdef __UDP_EXPERIMENTAL__
-                    m_checksumBad++;
-                    s.Printf("%d", m_checksumBad);
-                    m_txtChecksumBad->SetLabel(s);        
-#endif
-                }
-            }
-
-            //fprintf(g_logfile,"resetting callsign %s %ld\n", m_callsign, m_pcallsign-m_callsign);
-            // reset ptr to start of string
-            m_pcallsign = m_callsign;
-        }
-        else {
-            //fprintf(g_logfile, "new char %d %c\n", ashort, (char)ashort);
-            *m_pcallsign++ = (char)ashort;
+        unsigned char incomingChar = (unsigned char)ashort;
+        
+        if (wxGetApp().m_callsignEncoder)
+        {
+            wxGetApp().m_callsignEncoder->pushReceivedByte(incomingChar);    
+            if (wxGetApp().m_callsignEncoder->isInSync())
+            {
+                m_txtCtrlCallSign->SetValue(wxGetApp().m_callsignEncoder->getReceivedText());
+            }        
         }
-
-        /* If checksums disabled, display txt chars as they arrive */
-
-        if (!wxGetApp().m_enable_checksum) {
+        else
+        {
+            // Pre-1.5.1 behavior, where text is handled as-is.
+            if (incomingChar == '\r' || ((m_pcallsign - m_callsign) > MAX_CALLSIGN-1))
+            {                        
+                // CR completes line
+                *m_pcallsign = 0;
+                m_pcallsign = m_callsign;
+            }
+            else
+            {
+                *m_pcallsign++ = incomingChar;
+            }
             m_txtCtrlCallSign->SetValue(m_callsign);
         }
     }
 
-
+    // We should only report to PSK Reporter when all of the following are true:
+    // a) The callsign encoder indicates a valid callsign has been received.
+    // b) We detect a valid format callsign in the text (see https://en.wikipedia.org/wiki/Amateur_radio_call_signs).
+    // c) We don't currently have a pending report to add to the outbound list for the active callsign.
+    // When the above is true, capture the callsign and current SNR and save it in a temporary location.
+    // Once sync is lost, add to the PSK Reporter object's outbound list.
+    if (wxGetApp().m_pskReporter != NULL && wxGetApp().m_callsignEncoder != NULL)
```

Comment:
Good logic and ell worth documenting :+1: 

I can see is tweaking this over time as we get some real life experience with the feature.

Reply:
Taking another look, it looks like the logic is actually slightly off, so I'm going to clarify it in the lines below the comment (as well as update the comment).

Reply:
Actually, the code itself might be right but the comment still needs to be clarified. I'll update accordingly.